dbms/src/Functions/FunctionHelpers.h

```diff
@@ -24,6 +24,17 @@ const Type * checkAndGetDataType(const IDataType * data_type)
     return typeid_cast<const Type *>(data_type);
 }
 
+template <typename Type>
+std::shared_ptr<const Type> checkAndGetDataTypePtr(const DataTypePtr & data_type)
+{
+    if (typeid_cast<const Type *>(data_type.get()))
+    {
+        return std::static_pointer_cast<const Type>(data_type);
```

Comment:
`return nullptr` will do.

Reply:
removed that function altogether, since it isn't used anymore.